src/dataflow/src/sink/util.rs

```diff
@@ -7,28 +7,36 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
+use differential_dataflow::operators::arrange::ShutdownButton;
 use timely::dataflow::channels::pact::ParallelizationContract;
 use timely::dataflow::operators::generic::builder_rc::OperatorBuilder;
 use timely::dataflow::operators::generic::{FrontieredInputHandle, OperatorInfo};
 use timely::dataflow::{Scope, Stream};
 use timely::Data;
 
-pub fn sink_reschedule<G: Scope, D, B, L, P>(s: &Stream<G, D>, pact: P, name: &str, constructor: B)
+pub fn sink_reschedule<G: Scope, D, B, L, P, T>(
+    s: &Stream<G, D>,
+    pact: P,
+    name: &str,
+    constructor: B,
+) -> ShutdownButton<T>
 where
     D: Data,
-    B: FnOnce(OperatorInfo) -> L,
+    B: FnOnce(OperatorInfo) -> (L, ShutdownButton<T>),
     L: FnMut(&mut FrontieredInputHandle<G::Timestamp, D, P::Puller>) -> bool + 'static,
     P: ParallelizationContract<G::Timestamp, D>,
 {
     let mut builder = OperatorBuilder::new(name.to_owned(), s.scope());
     let operator_info = builder.operator_info();
     let mut input = builder.new_input(s, pact);
+    let (mut logic, button) = constructor(operator_info);
```

Comment:
note: I don't fully understand the consequences of lifting this function call out of the closure

Reply:
AFAICT it would only matter if `constructor` needed access to `capabilities`, but it does not.